packages/react-native/Package.swift

```diff
@@ -18,52 +19,577 @@ let cRNDepHeaderSearchPath: (Int) -> CSetting = {
+      extraGeneratedIncludePaths: ["Base", "Network", "ReactApple", "Required", "TypeSafety", "Modules", "CoreModules", "Views"]
+    ),
+
+    .reactNativeTarget(
+      name: .reactCoreModules,
+      dependencies: [.jsi],
+      path: "React/CoreModules",
+      extraExcludes: ["PlatformStubs/RCTStatusBarManager.mm"],
+      commonHeaderPathModules: ["ReactCommon", "ReactCommon/yoga", "React/FBReactNativeSpec", "Libraries/FBLazyVector", "ReactCommon/RuntimeExecutor", "ReactCommon/react/nativemodule/core/platform/ios", "ReactCommon/react/nativemodule/core", "ReactCommon/callinvoker"],
+      extraGeneratedIncludePaths: ["Base", "ReactApple", "Required", "TypeSafety", "CoreModules", "Views", "Modules", "Surface", "Surface_SurfaceHostingView", "Profiler", "DevSupport", "Inspector"]
+    ),
+
+    .reactNativeTarget(
+      name: .reactAppDelegate,
+      dependencies: [.jsi, .reactJsiExecutor, .reactRuntime, .reactRCTImage, .reactHermes, .reactCore, .reactFabric, .reactTurboModuleCore],
+      path: "Libraries/AppDelegate",
+      commonHeaderPathModules: ["ReactCommon", "ReactCommon/yoga", "ReactCommon/RuntimeExecutor", "ReactCommon/callinvoker", "ReactCommon/react/renderer/graphics/platform/ios", "ReactCommon/react/runtime/platform/ios", "ReactCommon/react/nativemodule/core", "ReactCommon/react/nativemodule/core/platform/ios", "ReactCommon/hermes", "ReactCommon/jsiexecutor"],
+      extraGeneratedIncludePaths: ["Base", "ReactApple", "Views", "CxxUtils", "Modules", "Fabric", "CoreModules", "CxxBridge", "Fabric_Mounting", "Fabric_Surface", "Surface", "Image", "Network", "Surface_SurfaceHostingView", "Fabric_Utils", "Runtime", "Hermes_Executor"]
+    ),
+
+    .reactNativeTarget(
+      name: .reactCodegen,
+      dependencies: [.reactJsiExecutor, .rctTypesafety, .reactCore, .jsi, .reactTurboModuleCore, .reactTurboModuleBridging, .reactRuntimeApple, .reactGraphics, .reactRendererDebug, .reactFabric, .reactDebug, .reactAppDelegate],
+      path: ".build/codegen/build/generated",
+      extraExcludes: ["ios/RCTAppDependencyProvider.h", "ios/RCTAppDependencyProvider.mm"],
+      commonHeaderPathModules: ["ReactCommon", "ReactCommon/yoga", "ReactCommon/RuntimeExecutor", "ReactCommon/callinvoker", "ReactCommon/react/renderer/graphics/platform/ios", "ReactCommon/react/runtime/platform/ios", "ReactCommon/react/nativemodule/core", "ReactCommon/react/nativemodule/core/platform/ios", "ReactCommon/hermes", "ReactCommon/jsiexecutor"],
+      extraGeneratedIncludePaths: ["Base", "ReactApple", "Views", "CxxUtils", "Modules", "Fabric", "CoreModules", "CxxBridge", "Fabric_Mounting", "Fabric_Surface", "Surface", "Image", "Network", "Surface_SurfaceHostingView", "Fabric_Utils", "Runtime", "Hermes_Executor", "ReactCodegen"]
     ),
```

Comment:
this can be removed. We don't have to build codegen as this is app specific. We still have to run codegen, because it also generates some files in `ReactCommon/react/renderer/components/rncore` and in `FBReactNativeSpec`